### DISCUSSION TITLE
Can sycamore views be loaded in dynamically?

### DISCUSSION THREAD

**Discussion by mainrs** [sycamore-rs/sycamore#609] [Q&A]
I want my application to be able to define custom UI elements as plugins. The user can install certain plugins and they then show up in my application.

Is it possible for sycamore components to be distributed via WASM as components themselves so I can load them dynamically into my application?

My project is a WYSIWYG editor for a custom data format. That format is extensible and I want new nodes that the user can drag & drop to be distributed as plugins.

**Comment by lukechu10** [comment]
Unfortunately this would be pretty hard to do because we don't have first-class support for wasm code splitting yet. I'm guessing you intend the plugins to be written using sycamore as well?

**Reply by mainrs** [reply]
> I'm guessing you intend the plugins to be written using sycamore as well?

Yes. The idea is to be able to do that. I chose `sycamore` since it did not use a VDOM. My UI uses `react-flow`. And the custom sycamore plugins would add custom node content to that node-based graph. Having a VDOM would not make this possible AFAIK. And thus I cannot use the likes of `dioxys` and `yew`.

**Reply by lukechu10** [reply]
If you're already using React Flow, why not just use React then? Otherwise you would be paying for the overhead of two UI frameworks at the same time.

Also it seems like React Flow expects custom nodes to be React components? If you were to build them with Sycamore, you would need to create wrappers around them as React components which honestly I think is more effort than its worth.

**Reply by mainrs** [reply]
The whole application is based on Tauri. I was hoping to move my current solution from `React` + `Rust` plugins (UI in JS, tauri commands in Rust) to Rust-only. My current system is super complicated and feels more hacky than anything else.

**Reply by lukechu10** [reply]
Is it absolutely necessary to dynamically load the plugins? Especially since this is a client side app, there should be no network delay so could you just ship everything in one WASM file?
